tm2/pkg/bft/abci/types/types.go

```diff
@@ -116,6 +117,32 @@ func (r ResponseBase) IsErr() bool {
 	return r.Error != nil
 }
 
+func (r ResponseBase) EventString() string {
+	levts := len(r.Events)
+	if levts == 0 {
+		return `""`
+	}
+
+	var builder strings.Builder
+
+	if levts > 1 {
+		builder.WriteString("[")
+	}
+
+	for i, event := range r.Events {
+		if i > 0 {
+			builder.WriteString(",")
+		}
+		builder.WriteString(event.String())
+	}
+
+	if levts > 1 {
+		builder.WriteString("]")
+	}
```

Comment:
I believe this appears inconsistent and resembles JSON but is invalid. How about making it more human-readable or ensuring it's valid JSON?

Reply:
I think it would be better to use the `type` that comes in as the _key_ in the JSON string, and put the remaining attributes and metadata such as timestamps as the _value_. By doing this, we can consistently represent the data and efficiently store multiple events.

```json
{
  "sender": {
    "PkgPath": "gno.land/r/demo/ee",
    "Identifier": "Hello",
    "Timestamp": 1713333669,
    "Attributes": [],
  },
  "receiver": {
    "PkgPath": "gno.land/r/demo/ee",
    "Identifier": "Hello",
    "Timestamp": 1713333669,
    "Attributes": [
       {
         "Key": "foo",
         "Value": "bar",
       }
     ],
   }
}
```